### PR TITLE
vscode: migrate to more type-safe assert impl

### PR DESCRIPTION
**Pull request by Veetaha**
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions

### CODE REVIEW

**Comment by Veetaha** [comment]
@lnicola, @matklad , why can't I add reviewers anymore???
image

File: editors/code/src/util.ts [hunk 1]
```diff
@@ -1,22 +1,41 @@
 import * as lc from "vscode-languageclient";
 import * as vscode from "vscode";
+import { strict as nativeAssert } from "assert";
+
+export function assert(condition: unknown, explanation: string): asserts condition {
```
Comment (matklad) [inline editors/code/src/util.ts]: I think a single `assert(condition: boolean, explanation: string)` should be enough.

**Comment by Veetaha** [comment]
@lnicola 
Though I am always for robustness so that we write the code that is statically strongly-typed and all-errors-cases-proven, for our VSCode this might not be the case. Protocol errors, sanity type checks ([1](https://github.com/rust-analyzer/rust-analyzer/blob/cc477edc6692608ca809ba39db274c5e837a3498/editors/code/src/config.ts#L144-L145), [2](https://github.com/rust-analyzer/rust-analyzer/blob/cc477edc6692608ca809ba39db274c5e837a3498/editors/code/src/installation/fetch_artifact_release_info.ts#L49)), naming invariants violation, hardware exhaustion (though I do have out-of hard-drive memory errors with the number of terabytes Rust debug builds occupy on my 68GB-space laptop), etc... handling all of these gracefully would bloat the codebase which we aspire to keep as small and concise as we can.

Even that `assert(Number.isNan(contentLength))` already seems excess as to me...

Anyway, we should be able to catch assertion errors and display them to the user. In fact `assert` in TypeScript doesn't crash the process, but instead throws `AssertionError`, just like we would 
```ts
throw new Error("Sanity check was not successful");
```

File: editors/code/src/util.ts (hunk 1 above)
Comment (Veetaha) [reply editors/code/src/util.ts]: Sure!

Comment (Veetaha) [reply editors/code/src/util.ts]: Sorry, do you mean `boolean`? If yes, we would need to explicitly use null comparisongs `assert(obj == null)` vs `assert(obj)`, or is this desired?

Comment (matklad) [reply editors/code/src/util.ts]: Yeah, I think explicit boolean is good

Comment (Veetaha) [reply editors/code/src/util.ts]: Done!

**Pull request merged**